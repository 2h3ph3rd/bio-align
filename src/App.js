import React from 'react'

import Grid from '@mui/material/Grid'
import Box from '@mui/material/Box'
import Paper from '@mui/material/Paper'

import AppHeader from './components/AppHeader.js'
import AppForm from './components/AppForm.js'
import AppTable from './components/AppTable.js'
import AppFooter from './components/AppFooter.js'

import { useTheme } from '@mui/material/styles'

export default function App() {
  const theme = useTheme()
  return (
    <>
      <Grid container spacing={2} sx={style.container}>
        <Grid item sx={offset(theme)} lg></Grid>
        <Grid item xs={12} md={4} lg={3} sx={style.centered}>
          <Paper sx={style.main} elevation={4}>
            <AppHeader></AppHeader>
            <AppForm></AppForm>
          </Paper>
        </Grid>
        <Grid item sx={offset(theme)} lg></Grid>
        <Grid item xs={12} md={8} lg={6} sx={style.centered}>
          <AppTable></AppTable>
        </Grid>
        <Grid item sx={offset(theme)} lg></Grid>
      </Grid>
      <Box sx={style.footer}>
        <AppFooter></AppFooter>
      </Box>
    </>
  )
}

const offset = (theme) => ({
  [theme.breakpoints.down('lg')]: {
    display: 'none',
  },
})

const style = {
  container: {
    minHeight: '100vh',
  },
  centered: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
  },
  main: {
    p: 3,
  },
  footer: {
    paddingBottom: '3vh',
    paddingLeft: '5vw',
    paddingRight: '5vw',
  },
}
