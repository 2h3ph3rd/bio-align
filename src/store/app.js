import { createSlice } from '@reduxjs/toolkit'

import logic from '../logic'

const defaultAlgorithm = logic.algorithms[0]

const defaultData = {
  algorithm: 0,
  firstString: 'ATTACTC',
  secondString: 'ATATGTC',
  ...defaultAlgorithm.defaultData,
}

const initialState = {
  form: defaultData,
  solution: defaultAlgorithm.calculate(defaultData),
  stringsMatch: defaultAlgorithm.stringsMatch(defaultData),
}

export const app = createSlice({
  name: 'counter',
  initialState,
  reducers: {
    setAlgorithm: (state, action) => {
      state.form.algorithm = action.payload
      const algorithm = logic.algorithms[state.form.algorithm]
      state.form = Object.assign(state.form, algorithm.defaultData)
      state.solution = recalculate(state)
    },
    setFirstString: (state, action) => {
      state.form.firstString = action.payload
      state.solution = recalculate(state)
    },
    setSecondString: (state, action) => {
      state.form.secondString = action.payload
      state.solution = recalculate(state)
    },
    setMatchWeight: (state, action) => {
      state.form.matchWeight = action.payload
      state.solution = recalculate(state)
    },
    setMismatchWeight: (state, action) => {
      state.form.mismatchWeight = action.payload
      state.solution = recalculate(state)
    },
    setGapWeight: (state, action) => {
      state.form.gapWeight = action.payload
      state.solution = recalculate(state)
    },
  },
})

function recalculate(state) {
  return logic.algorithms[state.form.algorithm].calculate(state.form)
}

export const {
  setAlgorithm,
  setFirstString,
  setSecondString,
  setMatchWeight,
  setMismatchWeight,
  setGapWeight,
} = app.actions

export default app.reducer
