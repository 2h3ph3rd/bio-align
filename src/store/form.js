import { createSlice } from '@reduxjs/toolkit'

import logic from '../logic'

const initialState = {
  algorithm: 0,
  firstString: 'ATTACTC',
  secondString: 'ATATGTC',
  matchWeight: logic.algorithms[0].matchWeight,
  mismatchWeight: logic.algorithms[0].mismatchWeight,
  gapWeight: logic.algorithms[0].gapWeight,
}

export const form = createSlice({
  name: 'counter',
  initialState,
  reducers: {
    setAlgorithm: (state, action) => {
      state.algorithm = action.payload
      state.matchWeight = logic.algorithms[state.algorithm].matchWeight
      state.mismatchWeight = logic.algorithms[state.algorithm].mismatchWeight
      state.gapWeight = logic.algorithms[state.algorithm].gapWeight
    },
    setFirstString: (state, action) => {
      state.firstString = action.payload
    },
    setSecondString: (state, action) => {
      state.secondString = action.payload
    },
    setMatchWeight: (state, action) => {
      state.matchWeight = action.payload
    },
    setMismatchWeight: (state, action) => {
      state.mismatchWeight = action.payload
    },
    setGapWeight: (state, action) => {
      state.gapWeight = action.payload
    },
  },
})

export const {
  setAlgorithm,
  setFirstString,
  setSecondString,
  setMatchWeight,
  setMismatchWeight,
  setGapWeight,
} = form.actions

export default form.reducer
