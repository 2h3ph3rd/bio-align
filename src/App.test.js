import React from 'react'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'

import App from './App'
import { store } from './store'

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <App></App>
      </Provider>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
