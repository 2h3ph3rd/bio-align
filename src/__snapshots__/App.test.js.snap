// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    className="MuiBox-root css-d8lyhh"
  >
    <div
      className="MuiBox-root css-i3pbo"
    >
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation12 css-1gp9424-MuiPaper-root"
      >
        <div
          className="MuiBox-root css-1xhj18k"
        >
          <div
            className="MuiAvatar-root MuiAvatar-circular css-zjb5mq-MuiAvatar-root"
          >
            <img
              alt="app icon"
              className="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
              src="/icon.png"
            />
          </div>
          <h1
            className="MuiTypography-root MuiTypography-h3 css-1xuzize-MuiTypography-root"
          >
            BioAlign
          </h1>
        </div>
      </div>
    </div>
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation12 css-1gp9424-MuiPaper-root"
    >
      <div
        className="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth css-1iledgx-MuiFormControl-root"
      >
        <label
          className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1kty9di-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink={true}
          id="algorithm-label"
        >
          Algorithm
        </label>
        <div
          className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedStart MuiSelect-root css-1q6at85-MuiInputBase-root-MuiOutlinedInput-root"
          onClick={[Function]}
        >
          <div
            className="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-ittuaa-MuiInputAdornment-root"
          >
            <span
              className="notranslate"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "&#8203;",
                }
              }
            />
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="FunctionsIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M18 4H6v2l6.5 6L6 18v2h12v-3h-7l5-5-5-5h7z"
              />
            </svg>
          </div>
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-labelledby="algorithm-label algorithm"
            className="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedStart css-bf1e45-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
            id="algorithm"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            role="button"
            tabIndex={0}
          >
            Edit Distance
          </div>
          <input
            aria-hidden={true}
            className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
            disabled={false}
            onAnimationStart={[Function]}
            onChange={[Function]}
            required={false}
            tabIndex={-1}
            value="edit"
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-1z7n62"
            >
              <span>
                Algorithm
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        className="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth css-1iledgx-MuiFormControl-root"
      >
        <div
          className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          data-testid="value-input"
        >
          <label
            className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1kty9di-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={true}
          >
            First word
          </label>
          <div
            className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedStart css-1q6at85-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <div
              className="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-ittuaa-MuiInputAdornment-root"
            >
              <span
                className="notranslate"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="EditIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                />
              </svg>
            </div>
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedStart css-1o9s3wi-MuiInputBase-input-MuiOutlinedInput-input"
              disabled={false}
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value="ATTACTC"
            />
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-1z7n62"
              >
                <span>
                  First word
                </span>
              </legend>
            </fieldset>
          </div>
          
        </div>
      </div>
      <div
        className="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth css-1iledgx-MuiFormControl-root"
      >
        <div
          className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          data-testid="value-input"
        >
          <label
            className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1kty9di-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={true}
          >
            Second word
          </label>
          <div
            className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedStart css-1q6at85-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <div
              className="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-ittuaa-MuiInputAdornment-root"
            >
              <span
                className="notranslate"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="EditIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                />
              </svg>
            </div>
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedStart css-1o9s3wi-MuiInputBase-input-MuiOutlinedInput-input"
              disabled={false}
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value="ATATGTC"
            />
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-1z7n62"
              >
                <span>
                  Second word
                </span>
              </legend>
            </fieldset>
          </div>
          
        </div>
      </div>
      <div
        className="MuiBox-root css-i3pbo"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-d3wcwz-MuiTypography-root"
          id="input-slider"
        >
          Weight
        </p>
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-ofrcno-MuiGrid-root"
        >
          <div
            className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="SwapHorizIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6.99 11 3 15l3.99 4v-3H14v-2H6.99v-3zM21 9l-3.99-4v3H10v2h7.01v3L21 9z"
              />
            </svg>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-4jz4pb-MuiGrid-root"
          >
            <span
              className="MuiSlider-colorPrimary MuiSlider-sizeMedium MuiSlider-root css-187mznn-MuiSlider-root"
              onMouseDown={[Function]}
            >
              <span
                className="MuiSlider-rail css-14pt78w-MuiSlider-rail"
              />
              <span
                className="MuiSlider-track css-1gv0vcd-MuiSlider-track"
                style={
                  Object {
                    "left": "0%",
                    "width": "50%",
                  }
                }
              />
              <span
                className="MuiSlider-thumbColorPrimary MuiSlider-thumbSizeMedium MuiSlider-thumb css-eg0mwd-MuiSlider-thumb"
                data-index={0}
                onMouseLeave={[Function]}
                onMouseOver={[Function]}
                style={
                  Object {
                    "left": "50%",
                    "pointerEvents": undefined,
                  }
                }
              >
                <input
                  aria-orientation="horizontal"
                  aria-valuemax={10}
                  aria-valuemin={-10}
                  aria-valuenow={0}
                  data-index={0}
                  disabled={false}
                  max={10}
                  min={-10}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  step={1}
                  style={
                    Object {
                      "border": 0,
                      "clip": "rect(0 0 0 0)",
                      "direction": "ltr",
                      "height": "100%",
                      "margin": -1,
                      "overflow": "hidden",
                      "padding": 0,
                      "position": "absolute",
                      "whiteSpace": "nowrap",
                      "width": "100%",
                    }
                  }
                  type="range"
                  value={0}
                />
              </span>
            </span>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <div
              className="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-sizeSmall css-18igo3j-MuiInputBase-root-MuiInput-root"
              onClick={[Function]}
              variant="outlined"
            >
              <input
                aria-labelledby="input-slider"
                className="MuiInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-nz481w-MuiInputBase-input-MuiInput-input"
                max={10}
                min={-10}
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                step={1}
                type="number"
                value={0}
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiBox-root css-i3pbo"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-d3wcwz-MuiTypography-root"
          id="input-slider"
        >
          Weight
        </p>
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-ofrcno-MuiGrid-root"
        >
          <div
            className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="SwapHorizIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6.99 11 3 15l3.99 4v-3H14v-2H6.99v-3zM21 9l-3.99-4v3H10v2h7.01v3L21 9z"
              />
            </svg>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-4jz4pb-MuiGrid-root"
          >
            <span
              className="MuiSlider-colorPrimary MuiSlider-sizeMedium MuiSlider-root css-187mznn-MuiSlider-root"
              onMouseDown={[Function]}
            >
              <span
                className="MuiSlider-rail css-14pt78w-MuiSlider-rail"
              />
              <span
                className="MuiSlider-track css-1gv0vcd-MuiSlider-track"
                style={
                  Object {
                    "left": "0%",
                    "width": "50%",
                  }
                }
              />
              <span
                className="MuiSlider-thumbColorPrimary MuiSlider-thumbSizeMedium MuiSlider-thumb css-eg0mwd-MuiSlider-thumb"
                data-index={0}
                onMouseLeave={[Function]}
                onMouseOver={[Function]}
                style={
                  Object {
                    "left": "50%",
                    "pointerEvents": undefined,
                  }
                }
              >
                <input
                  aria-orientation="horizontal"
                  aria-valuemax={10}
                  aria-valuemin={-10}
                  aria-valuenow={0}
                  data-index={0}
                  disabled={false}
                  max={10}
                  min={-10}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  step={1}
                  style={
                    Object {
                      "border": 0,
                      "clip": "rect(0 0 0 0)",
                      "direction": "ltr",
                      "height": "100%",
                      "margin": -1,
                      "overflow": "hidden",
                      "padding": 0,
                      "position": "absolute",
                      "whiteSpace": "nowrap",
                      "width": "100%",
                    }
                  }
                  type="range"
                  value={0}
                />
              </span>
            </span>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <div
              className="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-sizeSmall css-18igo3j-MuiInputBase-root-MuiInput-root"
              onClick={[Function]}
              variant="outlined"
            >
              <input
                aria-labelledby="input-slider"
                className="MuiInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-nz481w-MuiInputBase-input-MuiInput-input"
                max={10}
                min={-10}
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                step={1}
                type="number"
                value={0}
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiBox-root css-i3pbo"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-d3wcwz-MuiTypography-root"
          id="input-slider"
        >
          Weight
        </p>
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-ofrcno-MuiGrid-root"
        >
          <div
            className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="SwapHorizIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6.99 11 3 15l3.99 4v-3H14v-2H6.99v-3zM21 9l-3.99-4v3H10v2h7.01v3L21 9z"
              />
            </svg>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-4jz4pb-MuiGrid-root"
          >
            <span
              className="MuiSlider-colorPrimary MuiSlider-sizeMedium MuiSlider-root css-187mznn-MuiSlider-root"
              onMouseDown={[Function]}
            >
              <span
                className="MuiSlider-rail css-14pt78w-MuiSlider-rail"
              />
              <span
                className="MuiSlider-track css-1gv0vcd-MuiSlider-track"
                style={
                  Object {
                    "left": "0%",
                    "width": "50%",
                  }
                }
              />
              <span
                className="MuiSlider-thumbColorPrimary MuiSlider-thumbSizeMedium MuiSlider-thumb css-eg0mwd-MuiSlider-thumb"
                data-index={0}
                onMouseLeave={[Function]}
                onMouseOver={[Function]}
                style={
                  Object {
                    "left": "50%",
                    "pointerEvents": undefined,
                  }
                }
              >
                <input
                  aria-orientation="horizontal"
                  aria-valuemax={10}
                  aria-valuemin={-10}
                  aria-valuenow={0}
                  data-index={0}
                  disabled={false}
                  max={10}
                  min={-10}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  step={1}
                  style={
                    Object {
                      "border": 0,
                      "clip": "rect(0 0 0 0)",
                      "direction": "ltr",
                      "height": "100%",
                      "margin": -1,
                      "overflow": "hidden",
                      "padding": 0,
                      "position": "absolute",
                      "whiteSpace": "nowrap",
                      "width": "100%",
                    }
                  }
                  type="range"
                  value={0}
                />
              </span>
            </span>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <div
              className="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-sizeSmall css-18igo3j-MuiInputBase-root-MuiInput-root"
              onClick={[Function]}
              variant="outlined"
            >
              <input
                aria-labelledby="input-slider"
                className="MuiInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-nz481w-MuiInputBase-input-MuiInput-input"
                max={10}
                min={-10}
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                step={1}
                type="number"
                value={0}
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation12 css-1bywbq1-MuiPaper-root"
  >
    <div
      className="MuiBox-root css-1j9zy8k"
    >
      <div
        className="MuiBox-root css-0"
      />
      <div
        className="MuiBox-root css-xi606m"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 css-f46jb1-MuiTypography-root"
        >
          Made with 
          <a
            className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-1os7umh-MuiTypography-root-MuiLink-root"
            onBlur={[Function]}
            onFocus={[Function]}
          >
            ♥
          </a>
           using 
          <a
            className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-1oaaite-MuiTypography-root-MuiLink-root"
            href="https://reactjs.org/"
            onBlur={[Function]}
            onFocus={[Function]}
            target="_blank"
          >
            React.js
          </a>
           and 
          <a
            className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-1oaaite-MuiTypography-root-MuiLink-root"
            href="https://mui.com"
            onBlur={[Function]}
            onFocus={[Function]}
            target="_blank"
          >
            MUI
          </a>
        </p>
      </div>
      <div
        className="MuiBox-root css-s2uf1z"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-m5acjk-MuiSvgIcon-root"
          data-testid="GitHubIcon"
          focusable="false"
          onClick={[Function]}
          viewBox="0 0 24 24"
        >
          <path
            d="M12 1.27a11 11 0 00-3.48 21.46c.55.09.73-.28.73-.55v-1.84c-3.03.64-3.67-1.46-3.67-1.46-.55-1.29-1.28-1.65-1.28-1.65-.92-.65.1-.65.1-.65 1.1 0 1.73 1.1 1.73 1.1.92 1.65 2.57 1.2 3.21.92a2 2 0 01.64-1.47c-2.47-.27-5.04-1.19-5.04-5.5 0-1.1.46-2.1 1.2-2.84a3.76 3.76 0 010-2.93s.91-.28 3.11 1.1c1.8-.49 3.7-.49 5.5 0 2.1-1.38 3.02-1.1 3.02-1.1a3.76 3.76 0 010 2.93c.83.74 1.2 1.74 1.2 2.94 0 4.21-2.57 5.13-5.04 5.4.45.37.82.92.82 2.02v3.03c0 .27.1.64.73.55A11 11 0 0012 1.27"
          />
        </svg>
      </div>
    </div>
  </div>,
]
`;
